cai_causal_graph/causal_graph.py

```diff
@@ -1588,15 +1606,9 @@ def get_ancestral_graph(self, node: NodeLike) -> CausalGraph:
         Get a sub causal graph that only includes the ancestors of a specific node and the node itself.
         """
         identifier = self._NodeCls.identifier_from(node)
-        ancestors: List[str] = [*self.get_ancestors(node), identifier]
+        ancestors = [*self.get_ancestors(identifier), identifier]
 
-        ancestral_graph = self.copy()
-
-        for i in ancestral_graph.nodes:
-            if i.identifier not in ancestors:
-                ancestral_graph.delete_node(i.identifier)
-
-        return ancestral_graph
+        return self._filter_graph_by_nodes(ancestors)   # type: ignore
```

Comment:
why do you need to ignore the type hinting here?

Reply:
Mypy was throwing an issue around List being invariant with the previous List[str] typing. I have updated it to just do better typing as opposed to removing the old typing